core/src/main/java/org/infinispan/xsite/irac/DefaultIracManager.java

```diff
@@ -102,10 +114,14 @@
    private final LongAdder conflictMergedCount = new LongAdder();
 
    public DefaultIracManager(Configuration config) {
-      updatedKeys = new ConcurrentHashMap<>();
+      updatedKeys = new ConcurrentHashMap<>(64);
       iracExecutor = new IracExecutor(this::run);
       asyncBackups = asyncBackups(config);
-      setStatisticsEnabled(config.statistics().enabled());
+      statisticsEnabled = config.statistics().enabled();
+      batchSize = config.sites().asyncBackupsStream()
+            .map(BackupConfiguration::stateTransfer)
+            .map(XSiteStateTransferConfiguration::chunkSize)
```

Comment:
```suggestion
            .mapToInt(XSiteStateTransferConfiguration::chunkSize)
```
Just one last little thing, sorry didn't notice before.

Reply:
done